### PR TITLE
Update truncated tube shape parameters to match new version of DD4hep

### PR DESCRIPTION
**Pull request by cvuosalo**
DD4hep recently changed the parameters returned for describing the dimensions of a truncated tube. Instead of returning 16 parameters, DD4hep now returns 8 parameters. The CMS code had to be updated to read the new parameters correctly.

#### PR validation:

This new code was verified by running the magnetic field geometry builder, which uses truncated tubes. With this PR, the geometry builder builds and checks thousands of truncated tubes of various sizes with no errors.

No backport is needed.

### CODE REVIEW

**Comment by cvuosalo** [comment]
type bug-fix

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
@namapane: This PR fixes the bug with truncated tubes.

**Comment by cvuosalo** [comment]
+1

File: DetectorDescription/DDCMS/src/DDShapes.cc [hunk 1]
```diff
@@ -134,62 +134,32 @@ DDTruncTubs::DDTruncTubs(const DDFilteredView &fv) : valid{fv.isATruncTube()} {
...
-    const TGeoMatrix *rmatrix = boolNode->GetRightMatrix();
-    edm::LogVerbatim("DDShapes DDTruncTubs") << "Right matrix";
-    showMats(rmatrix);
+    startPhi_ = params[3];
+    deltaPhi_ = params[4];
+    cutAtStart_ = params[5];
+    cutAtDelta_ = params[6];
+    cutInside_ = (params[7] != 0);
+
+    /* The git history of this file from before 2019-11-25 has code for converting
```
Comment (fabiocos) [inline DetectorDescription/DDCMS/src/DDShapes.cc]: @cvuosalo could you please clarify this part? Is the change in DD4hep such that the conversion above is no more needed?

Comment (cvuosalo) [reply DetectorDescription/DDCMS/src/DDShapes.cc]: @fabiocos  I clarified the comment to say that the old complex conversion is not needed anymore.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
+1

@cvuosalo thanks for the clarification

**Pull request merged**